src/Audit.NET.AmazonQLDB/Providers/AmazonQldbDataProvider.cs

```diff
@@ -99,45 +113,32 @@ public override object InsertEvent(AuditEvent auditEvent)
         }
 
         /// <summary>
-        /// Asynchronously inserts an event into AmazonQLDB
+        /// Replaces an event into AmazonQLDB
         /// </summary>
-        public override Task<object> InsertEventAsync(AuditEvent auditEvent)
-        {
-            var result = InsertEvent(auditEvent);
-            return Task.FromResult(result);
-        }
+        public override void ReplaceEvent(object eventId, AuditEvent auditEvent) => ReplaceEventAsync(eventId, auditEvent).GetAwaiter().GetResult();
```

Comment:
`.GetResult()` can reasult in deadlock. 
We need a dealock-free version for synchronous context.
We could evaluate providing the option to use both an Async and/or Sync driver?

Reply:
The sync Amazon QLDB implementation also blocks on async code, so we would get into the same issue.

https://github.com/awslabs/amazon-qldb-driver-dotnet/blob/2e3cf8e2d635801732fcc5e5f4b82a9510ccaa38/Amazon.QLDB.Driver/client/Session.cs#L235